transformer_lens/HookedSAE.py

```diff
@@ -0,0 +1,95 @@
+from torch import nn
+import torch
+import torch.nn.functional as F
+import einops
+from transformer_lens.hook_points import (
+    HookedRootModule,
+    HookPoint,
+)  # Hooking utilities
+from transformer_lens.HookedSAEConfig import HookedSAEConfig
+
+from typing import Union, Dict
+
+
+class HookedSAE(HookedRootModule):
+    """Hooked AutoEncoder.
+    
+    Implements a standard SAE with a TransformerLens hooks for SAE activations
+    
+    Should probably just be used for inference / analysis, not training.
+    
+    This is just a template, feel free to copy this into a notebook and change it!
```

Comment:
What does this mean? What would you edit? The SAE architecture? I suggest removing the comment

Reply:
I removed it. But i do think it's somewhat important that people know they can hack the architecture and still work with HookedSAETransformer. It would be sad if someone couldn't use the library because their SAEs didn't use a pre-encoder bias, a different activation function, etc. I explained this a bit more in the "Attach SAEs" section of the [demo](https://colab.research.google.com/github/ckkissane/TransformerLens/blob/hooked-sae-transformer/demos/HookedSAETransformerDemo.ipynb#scrollTo=t3q9DzPZY6lX)

Reply:
I think it's worth saying this in the code! I would just reword a bit. Maybe:

"Note that HookedSAETransformer is fairly modular, and doesn't make strong assumptions about the architecture of the SAEs that get attached. We provide HookedSAE as a useful default class, but if you want to eg experiment with other SAE architectures, you can just copy the HookedSAE code, edit it, and add instances of the new SAE class to a HookedSAETransformer". Even better if you can enumerate what assumptions it does make.